providers/porkbun/api.go

```diff
@@ -150,3 +167,21 @@ func (c *porkbunProvider) updateNameservers(ns []string, domain string) error {
 	}
 	return nil
 }
+
+func (c *porkbunProvider) listAllDomains() ([]string, error) {
+	params := requestParams{}
+	var bodyString, err = c.post("/domain/listAll", params)
```

Comment:
This looks correct. However, experience tells me I should always ask this question:

Does Porkbun have "zones" and "domains"?  

A domain is something you can register.  Like... example.com or foo.uk.co.  A zone is a DNS zone (list of records).  For example, I registered the domain yesthattom.com but I have a zone called corp.yesthattom.com which is a subdomain (or a sub-zone).

I ask because ListZones() should return zones, not just domains.  Some APIs call them both "domains", others have different API calls for zones but refer to `/domains` only when dealing with registrations.



Thanks!

Reply:
The [Porkbun API documentation](https://porkbun.com/api/json/v3/documentation)  has 0 matches when searching for "zone".

Reply:
The [Porkbun API documentation](https://porkbun.com/api/json/v3/documentation)  has 0 matches when searching for "zone".

Reply:
Porkbun has both "zones" and "domains".

The worst thing is Porkbun combine them in this one `/domain/listAll` endpoint, and it is tricky to tell which one is zone, and which one is domain.

Reply:
Ok!  Sounds like you've already taken care of this.  Thanks for verifying!